_problems/secure-dlcs.md

```diff
@@ -0,0 +1,84 @@
+
+### Security Notions
+
+To be secure a DLC protocol must satisfy the following informal properties:
+
+1. Bet security: If the oracle or oracles are honest then the coins are distributed at the end of the protocol according to the agreement made between the two users.
+2. Oracle accountability: Any user wronged by a malicious oracle must be able to hold them accountable. It must be impossible to hold an honest oracle to account for something they didn't do.
+
+Note in the [original work][Discreet Log Contracts], a strong notion of oracle accountability is put forward such that a user obtains attestations for two contradictory outcomes is able to extract the oracle's static public key from it.
+
+### Attacks on DLCs
+
+So far the following attacks have been demonstrated or conjectured to exist against the original work:
+
+- **User-on-User rogue key attack**: A user can pick their public key to cancel out the oracle's attestation so that it doesn't need the oracle's attestation to claim the funds.[^1]
+- **Oracle-on-Oracle rogue key attack**: An oracle can choose their announcement nonce for an event to cancel out an attestation point of another oracle such that if the two are events are combined then the malicious oracle will be able to produce the combined attestation by itself.[^1]
+- **Oracle signature forgery**: A malicious party who has control over what the oracle attests to may get them to attest to a combination of outcomes such that they can forge a Schnorr signature by combining the attestations. They do this by solving a k-sum problem such that the sum of the Schnorr hashes equals their target value (note they can compute the Schnorr hashes ahead of time because the oracle includes the nonce in the announcement).[^2]
+- **Fraud proof forgery by collision**: A malicious user may be find two sets of different attestation points for the same set of announcements that add up to the same thing. Therefore, if the oracle honestly attests to one combination they will have implicitly attested to the other. The user may then present the attestation for the malicious combination as proof of fraud. Once again, this employs solving the k-sum problem[^2].
+
+## Impact
+
+The existing [DLC specification] effort has been impeded by having to consider security matters while
+A scheme with security proofs would allow users and engineers to be confident in the construction of predication markets and financial contracts on Bitcoin.
+
+## Potential Directions
+
+<!-- - The main use of listing hand-wavy directions is useful to further explore the problem. -->
+*There are no speculative directions towards solving this problem*
+
+## Proposed Solutions
+
+*There are no proposed solutions for this problem*
+
+## Related Research
+
+1. Dryja introduces the idea in *[Discreet Log Contracts]*.
+2. Fournier demonstrates the User-on-User rogue key attack in *[How to Make a Prediction Market on Twitter with Bitcoin]*.
```

Comment:
micronit for the history books: This was actually pointed out on #bitcoin-wizards almost immediately (June 6 2017), but Dryja apparently never bothered to update the paper.

Reply:
Ahhh thanks! That's good to know. Found it here: https://freenode.irclog.whitequark.org/bitcoin-wizards/2017-06-06. I'll add a footnote.